bindings/matrix-sdk-ffi/src/notification_settings.rs

```diff
@@ -0,0 +1,213 @@
+use std::sync::Arc;
+
+use anyhow::Context;
+use matrix_sdk::{
+    notification_settings::RoomNotificationMode as SdkRoomNotificationMode,
+    ruma::events::push_rules::PushRulesEvent, Client as MatrixClient,
+};
+use ruma::{push::Ruleset, RoomId};
+use tokio::sync::RwLock;
+
+use crate::error::NotificationSettingsError;
+
+#[derive(Clone, uniffi::Enum)]
+pub enum RoomNotificationMode {
+    AllMessages,
+    MentionsAndKeywordsOnly,
+    Mute,
+}
+
+impl From<SdkRoomNotificationMode> for RoomNotificationMode {
+    fn from(value: SdkRoomNotificationMode) -> Self {
+        match value {
+            SdkRoomNotificationMode::AllMessages => Self::AllMessages,
+            SdkRoomNotificationMode::MentionsAndKeywordsOnly => Self::MentionsAndKeywordsOnly,
+            SdkRoomNotificationMode::Mute => Self::Mute,
+        }
+    }
+}
+
+pub trait NotificationSettingsDelegate: Sync + Send {
+    fn notification_settings_did_change(&self);
+}
+
+#[derive(Clone, uniffi::Record)]
+pub struct RoomNotificationSettings {
+    mode: RoomNotificationMode,
+    is_default: bool,
+}
+
+impl RoomNotificationSettings {
+    fn new(mode: RoomNotificationMode, is_default: bool) -> Self {
+        RoomNotificationSettings { mode, is_default }
+    }
+}
+
+#[derive(Clone, uniffi::Object)]
+pub struct NotificationSettings {
+    pub(crate) sdk_client: MatrixClient,
+    push_rules: Arc<RwLock<Ruleset>>,
+    delegate: Arc<RwLock<Option<Box<dyn NotificationSettingsDelegate>>>>,
+}
+
+impl NotificationSettings {
+    pub(crate) fn new(sdk_client: MatrixClient) -> Self {
+        let push_rules = Arc::new(tokio::sync::RwLock::new(Ruleset::new()));
+        let delegate: Arc<RwLock<Option<Box<dyn NotificationSettingsDelegate>>>> =
+            Arc::new(RwLock::new(None));
+
+        // Listen for PushRulesEvent
+        let push_rules_clone = push_rules.to_owned();
+        let delegate_clone = delegate.to_owned();
+        sdk_client.add_event_handler(move |ev: PushRulesEvent| {
```

Comment:
Is this sufficient to get all the push rules since the creation of the room? are we sure at least one `PushRulesEvent` is handled?